### PR TITLE
Fix and refactor browser page provenance issue

### PR DESCRIPTION
**Pull request by shifucun**
The name of a provenance triple is changed from url to the import name. Browser fetches all provenance and url, so the logic needs to be updated to fetch the url.

This PR also updates to use V1 API for triples fetch. This is only used in browser page.

### CODE REVIEW

File: server/routes/api/browser.py [hunk 1]
```diff
@@ -20,21 +20,34 @@
...
 
-@cache.memoize(timeout=3600 * 24)  # Cache for one day.
-@bp.route('/triples/<path:dcid>')
-def triple_api(dcid):
+@bp.route('/triples/<path:direction>/<path:dcid>')
+def triples(direction, dcid):
     """Returns all the triples given a node dcid."""
-    return json.dumps(dc.get_triples([dcid]).get(dcid, []))
+    if direction != "in" and direction != "out":
+        return f"Invalid direction {direction}, use 'in' or 'out'", 400
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/api/browser.py]: Done

File: server/routes/api/browser.py [hunk 2]
```diff
@@ -20,21 +20,34 @@
...
+    return dc.triples(dcid, direction).get("triples")
+
+
+@bp.route('/provenance')
+def provenance():
+    """Returns all the provenance information."""
+    resp = dc.triples("Provenance", "in")
+    prov_list = resp.get("triples").get("typeOf").get("nodes")
```
Comment (beets) [inline server/routes/api/browser.py:43]: return default objects if the keys aren't in the response?
```suggestion
    prov_list = resp.get("triples", {}).get("typeOf", {}).get("nodes", [])
```

File: server/routes/api/browser.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/api/browser.py]: Done

File: static/js/browser/browser_page.tsx [hunk 3]
```diff
@@ -176,31 +176,21 @@ export class BrowserPage extends React.Component<
...
       .then(([labelsData, ProvenanceData]) => {
         const provDomain = {};
-        for (const prov of ProvenanceData) {
-          if (this.isProvenanceEntity(prov)) {
-            try {
-              provDomain[prov["subjectId"]] = new URL(prov["subjectName"]).host;
-            } catch (err) {
-              console.log("Invalid url in prov: " + prov["subjectName"]);
-            }
+        for (const prov in ProvenanceData) {
```
Comment (beets) [inline static/js/browser/browser_page.tsx:189]: nit: while we're here, can we fix the var casing? s/ProvenanceData/provenanceData/

**Review by beets — APPROVED** [review]
thanks for the update - just a few access issues..

File: static/js/browser/browser_page.tsx (hunk 3 above)
Comment (shifucun) [reply static/js/browser/browser_page.tsx]: Done

File: server/routes/api/browser.py (hunk 2 above)
Comment (shifucun) [reply server/routes/api/browser.py]: Done. The data here is fixed (for all provenance), but changed to make it more defensive

**Review by shifucun — COMMENTED** [review]
Thanks for review!

**Pull request merged**